snap-core/src/main/java/org/esa/snap/core/dataio/geocoding/util/RasterUtils.java

```diff
@@ -219,7 +219,7 @@ public static double computeResolutionInKm(Band lonBand, Band latBand) {
         return computeResolutionInKm(resLons, resLats, R.width, R.height);
     }
 
-    private static double computeResolutionInKm(double[] lonData, double[] latData, int width, int height) {
+    public static double computeResolutionInKm(double[] lonData, double[] latData, int width, int height) {
```

Comment:
This method should remain private, as it efficiently calculates with small amounts of data, but would calculate with large amounts of data for a very long time.
In the calling method, the data sets are restricted to a field of 10x10 pixels maximum.